### DISCUSSION TITLE
Using RBFKernelGrad for 2D output

### DISCUSSION THREAD

**Discussion by meakbiyik** [cornellius-gp/gpytorch#2085] [Q&A]
Hi all!

I am trying to do some regression over 1D input to 2D output with a GP model. I also happen to have gradient information for the output dimensions, so the **x** is 1D (let's call it `t` as in time) but the **y** is 4D, with dimensions `a, b, da/dt, db/dt`.

I want to model this as a multitask GP with gradient information (so both the gradients and dimensions are modeled together), but I am not sure if this is even possible with GPyTorch. Here's an unsuccessful attempt that I got:

```python
class MultiTaskGPModelWithDerivatives(gpytorch.models.ExactGP):
    def __init__(self, train_x, train_y, likelihood):
        super(MultiTaskGPModelWithDerivatives, self).__init__(train_x, train_y, likelihood)
        self.mean_module = gpytorch.means.MultitaskMean(
            gpytorch.means.ConstantMeanGrad(), num_tasks=2
        )
        self.covar_module = gpytorch.kernels.MultitaskKernel(
            gpytorch.kernels.RBFKernelGrad(), num_tasks=2, rank=2
        )

    def forward(self, x):
        mean_x = self.mean_module(x)
        covar_x = self.covar_module(x)
        return gpytorch.distributions.MultitaskMultivariateNormal(mean_x, covar_x)
```

with likelihood
```python
gpytorch.likelihoods.MultitaskGaussianLikelihood(num_tasks=4)
```

Any help is appreciated!

**Comment by gpleiss** [comment]
I don't know whether anyone has tried multitask + derivatives before. In theory it should be possible, but it might be something that would require some internal changes.

cc/ @dme65
